### PR TITLE
Remove Party.Predicated

### PR DESCRIPTION
**Pull request by deepthiskumar**
and move `account_precondition` into `Party.Body`
Propagated the changes to the archive functionality as well.
Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #0000

### CODE REVIEW

File: src/app/archive/zkapp_tables.sql [hunk 1]
```diff
@@ -210,12 +210,12 @@ CREATE TABLE zkapp_party_body
 , call_data_id                          int        NOT NULL REFERENCES zkapp_state_data(id)
 , call_depth                            int        NOT NULL
 , zkapp_protocol_state_precondition_id     int        NOT NULL REFERENCES zkapp_protocol_state_precondition(id)
+, account_precondition_id             int                             NOT NULL REFERENCES zkapp_account_precondition(id)
```
Comment (mitschabaude) [inline src/app/archive/zkapp_tables.sql:213]: nit: the column spacing got messed up here

File: src/lib/mina_base/party.ml [hunk 2]
```diff
@@ -851,343 +992,53 @@ module Body = struct
...
-    module Stable = struct
-      module V1 = struct
-        type ('body, 'account_precondition) t =
-          { body : 'body; account_precondition : 'account_precondition }
-        [@@deriving annot, hlist, sexp, equal, yojson, hash, compare, fields]
-      end
-    end]
-  end
-
+module T = struct
```
Comment (mitschabaude) [inline src/lib/mina_base/party.ml:999]: Curious why you had to add a wrapping `T`module here that's included later!

File: src/lib/snarky_js_bindings/lib/snarky_js_bindings_lib.ml [hunk 3]
```diff
@@ -2455,7 +2455,7 @@ module Ledger = struct
   (* TODO hash two parties together in the correct way *)
 
   let hash_party (p : party) =
-    p |> party |> Party.Preconditioned.digest |> Field.constant |> to_js_field
+    p |> party |> Party.digest |> Field.constant |> to_js_field
```
Comment (mitschabaude) [inline src/lib/snarky_js_bindings/lib/snarky_js_bindings_lib.ml:2458]: There's a lot of code in here that still needs fixing... like the `body` and `party` functions above (checked + unchecked versions).

**Review by mitschabaude — APPROVED** [review]
Very nice change @deepthiskumar! There are still a couple TODOs in snarky_js_bindings_lib that CI will fail on, otherwise LGTM

**Review by mrmr1993 — APPROVED** [review]

File: src/lib/mina_base/party.ml (hunk 2 above)
Comment (deepthiskumar) [reply src/lib/mina_base/party.ml:999]: To specify the return type in `Fee_payer.to_party`

File: src/lib/snarky_js_bindings/lib/snarky_js_bindings_lib.ml [hunk 4]
```diff
@@ -1778,19 +1801,11 @@ module Ledger = struct
...
-    type t =
-      < type_ : Js.js_string Js.t Js.prop ; value : party_predicate Js.prop >
-      Js.t
-  end
-
-  type party =
-    < body : party_body Js.prop ; predicate : Party_predicate.t Js.prop > Js.t
+  (*Reviewer: Leave this structure as is for authorization to be added later or
+    should it be an alias of party_body?*)
+  type party = < body : party_body Js.prop > Js.t
```
Comment (mitschabaude) [inline src/lib/snarky_js_bindings/lib/snarky_js_bindings_lib.ml:1806]: LGTM, let's leave it like that for now

**Pull request merged**